storage/local/storage.go

```diff
@@ -510,6 +522,11 @@ func (s *memorySeriesStorage) DropMetricsForFingerprints(fps ...clientmodel.Fing
 
 // Append implements Storage.
 func (s *memorySeriesStorage) Append(sample *clientmodel.Sample) {
+	for ln, lv := range sample.Metric {
+		if len(lv) == 0 {
+			delete(sample.Metric, ln)
+		}
+	}
```

Comment:
Since Go has no const guarantees, I'd always document whether something passed in with a pointer will be modified (or in case of an interface, if the implementation is allowed to modify it).

Something like:

"Implementations (of Append) might remove labels with empty values from sample as those are considered equivalent to a label not present at all."